### DISCUSSION TITLE
Can Bromite Be Started With This Switch? (Enables Viewing of Self-Extracting ZIP/HTML Files)

### DISCUSSION THREAD

**Discussion by OkyDooky** [bromite/bromite#2501] [Q&A] [answered]
Since extensions/add-ons cannot be used in Bromite, I was wondering if it is possible to use the following procedure to be able to view files made with [SingleFileZ](https://github.com/gildas-lormeau/SingleFileZ):
`Otherwise, the browser must be started with the switch --allow-file-access-from-files.`
I am unsure how to do this, since I am not a Chromium user (aside from Bromite) and only have mild technical experience with Firefox-based browsers.
In case someone wants to suggest "just use the built-in MHTML features," I'd like to be able to use a more universally accepted method and format. No approach is truly universal, it seems, but it's shocking how limited the support is for MHTML, despite being around for so long. You can see this in the [Wikipedia list](https://en.wikipedia.org/wiki/MHTML#Browser_support) and some discussion on the [Chromium bug tracker](https://bugs.chromium.org/p/chromium/issues/detail?id=1235248).
There's also a lengthy and interesting discussion on formats and various approaches found in [this issue on SFZ's Github](https://github.com/gildas-lormeau/SingleFileZ/issues/47) where they discuss the workings and merits of the MHTML, .zip.hmt(l), and MAFF formats.
Here's [another supplemental discussion](https://github.com/gildas-lormeau/SingleFileZ/issues/111).

I'm not discounting the inclusion of MHTML features as much as I'm asking for the ability to also open this format in Bromite. I was making an issue for this, but realized I should ask if I can toggle it myself before asking the developers to do it for me or add it as a toggle in the settings, somewhere.

Thanks in advanced.

**Comment by OkyDooky** [comment]
Thanks for bumping my post, bot.

**Comment by csagan5** [comment]
Does this format work with the desktop version of Chromium? If not it will also not work in Bromite.

**Reply by OkyDooky** [reply]
Supposedly, yes (otherwise, I wouldn't be asking). If Chromium has the extension installed, it should work out of the box. But, if it doesn't the flag I referenced is apparently needed for it to be willing to open them.
The first link in my OP gives more detail about how this works.

The issue is that this is aimed at the desktop version and I don't see how or where I would be able to do that on Android. If it's something I can do, then great. If not, I would have to ask the developers nicely if they would implement a way to enable it that doesn't violate any of Bromite's principles.

**Comment by Universalizer** [comment, marked as answer]
#### Guess work (not sure), under chrome://flags/

Required flags, might be helpful,<hr />

#### Enable Local File Accesses

Enable local file accesses. Useful for web development. – Mac, Windows, Linux, ChromeOS, Android, Fuchsia, Lacros

#enable-local-file-accesses<hr />

#### Allow File URI Access from Files

By default, file:// URIs cannot read other file:// URIs. This is an override for web developers who need the old behavior for testing. – Mac, Windows, Linux, ChromeOS, Android, Fuchsia, Lacros

#allow-file-access-from-files

**Reply by OkyDooky** [reply]
Uazo's build of Chromium doesn't include that flag and neither does the last stable build of Bromite. Thorium's Android/Arm build does include it and it can read the files, so thank you.
Hopefully, this feature can be brought to Bromite, as well.

**Comment by Universalizer** [comment]
Many thanks to @Alex313031 for "Thorium-Special" architecture build "arm32 & 64", version 112.0.5615.166 (561516610) in which i found these,

Didn't recognize in previous version.

Edit: Alex, if you could retain these flags permanently, in your arm builds.👍

**Reply by Alex313031** [reply]
Aww thanks
I try lol
